pkg/query/path.go

```diff
@@ -206,6 +206,72 @@ func MustPathFromString(relationshipStr string) *Path {
 	return FromRelationship(rel)
 }
 
+// EqualsEndpoints checks if two paths have the same Resource and Subject endpoints (types and IDs only)
+func (p *Path) EqualsEndpoints(other *Path) bool {
+	if p == nil || other == nil {
+		return p == other
+	}
+
+	return p.Resource.ObjectType == other.Resource.ObjectType &&
+		p.Resource.ObjectID == other.Resource.ObjectID &&
+		p.Subject.ObjectType == other.Subject.ObjectType &&
+		p.Subject.ObjectID == other.Subject.ObjectID
+}
+
+// Equals checks if two paths are fully equal (all fields match)
+func (p *Path) Equals(other *Path) bool {
```

Comment:
Oof. Are there libs for this kind of thing? I'm envisioning something that lets you implement something like a python dunder but that has some kind of default for fieldwise comparison.